src/material.js

```diff
@@ -173,20 +173,42 @@ export class Material {
     }
 
     /**
-     * Calculates hash from basis and lattice. Algorithm expects the following:
+     * Calculates the hash for a material.
+     * If `isNonPeriodic` = false
+     * The hash is calculated from the basis and lattice. The algorithm expects the following:
      * - asserts lattice units to be angstrom
      * - asserts basis units to be crystal
      * - asserts basis coordinates and lattice measurements are rounded to hash precision
      * - forms strings for lattice and basis
      * - creates MD5 hash from basisStr + latticeStr + salt
+     *
+     * If `isNonPeriodic` = true
+     * The hash based on the `inchi` string for the material that is returned from getNonPeriodicHashMessage.
+     *
      * @param salt {String} Salt for hashing, empty string by default.
      * @param isScaled {Boolean} Whether to scale the lattice parameter 'a' to 1.
      */
     calculateHash(salt = '', isScaled = false) {
-        const message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        let message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        if (this._json.isNonPeriodic) {
+            message = this.getNonPeriodicHashMessage() || message;
+        }
         return CryptoJS.MD5(message).toString();
     }
 
+    /**
+     * Gets the `inchi` string from the derived properties and returns it as the message to use for
+     * generating the hash when `isNonPeriodic` = true.
+     * @returns String
+     */
+    getNonPeriodicHashMessage() {
+        const derivedProperties = this._json.derivedProperties;
+        const inchi = lodash.isArray(derivedProperties) ? derivedProperties.find(x => x.name === "inchi") : null;
+        if (inchi) {
+            return inchi.value;
```

Comment:
Should we only use the inchi, or should we include basis information as well? I think the inchi should suffice. Adding in basis information could cause the same material to be generated with two different hashes if the basis differs. 

The only reason I could see to include the basis would be if we want a way to distinguish between conformers of the same structure, but at this stage I do not know if that is important. Conformers are something many computational chemists still struggle with handling and uniquely identifying properly, especially for acyclic structures. It's not a trivial task to say the least.

 I do not see any reason really to include the lattice.

Reply:
I agree with what you wrote above, but the implementation needs to be adjusted